kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala

```diff
@@ -108,4 +109,19 @@ object HighAvailabilityConf {
     .version("1.4.0")
     .stringConf
     .createOptional
+
+  val HA_ZK_ENGINE_POOL_SIZE: ConfigEntry[Int] =
+    buildConf("ha.engine.pool.size")
+      .doc("Maximum number of engines when using USER share level.")
+      .version("1.4.0")
+      .intConf
+      .createWithDefault(1)
```

Comment:
How about use `checkValues` to make sure this configuration remains positive?

Reply:
Make sense, thanks.